projects/addon-editor/extensions/file-link/file-link.ts

```diff
@@ -0,0 +1,43 @@
+import {TuiEditorAttachedFile} from '@taiga-ui/addon-editor/interfaces';
+import {Extension} from '@tiptap/core';
+
+declare module '@tiptap/core' {
+    interface Commands<ReturnType> {
+        fileLink: {
+            /**
+             * Comments will be added to the autocomplete.
+             */
+            setFileLink: (preview: TuiEditorAttachedFile) => ReturnType;
+        };
+    }
+}
+
+export const TuiFileLink = Extension.create({
+    name: `fileLink`,
+
+    addCommands() {
+        return {
+            setFileLink:
+                fileLink =>
+                ({state, chain}) => {
+                    const {selection} = state;
+                    const selectedSize = Math.abs(selection.to - selection.from);
+
+                    return (
+                        selectedSize > 0
+                            ? chain().toggleMark(
+                                  `link`,
+                                  {href: fileLink.link},
+                                  {extendEmptyMarkRange: true},
+                              )
+                            : chain().insertContent(
+                                  `<a rel="noopener noreferrer nofollow" href="${fileLink.link}">${fileLink.name}</a><span style="font-size: 15px"> </span>`,
```

Comment:
For some reason it looks bloated in height:
![image](https://user-images.githubusercontent.com/11832552/201710651-0916cffb-1b7e-4423-a8c9-27be7656d290.png)